test/stdlib/Integers.swift.gyb

```diff
@@ -825,6 +825,35 @@ tests.test("isMultiple") {
   isMultipleTest(type: UInt64.self)
 }
 
+tests.test("isPower") {
+  func isPowerTest<T: FixedWidthInteger>(type: T.Type) {
+    expectTrue((0 as T).isPower(of: 0))
+    expectTrue((1 as T).isPower(of: 0))
+    expectTrue((100 as T).isPower(of: 10))
+    expectFalse(T.max.isPower(of: 2))
+    expectFalse(T.min.isPower(of: 2))
+    if T.isSigned {
+      expectTrue((1 as T).isPower(of: -1))
+      expectTrue((-1 as T).isPower(of: -1))
+      expectFalse((-1 as T).isPower(of: 1))
+      expectTrue(T.min.isPower(of: -2))
+      expectTrue(((1 as T) << (T.bitWidth - 2)).isPower(of: 2))
+    } else {
+      expectTrue(((1 as T) << (T.bitWidth - 1)).isPower(of: 2))
+    }
+  }
```

Comment:
Should this test more values? For example:

```swift
  for value in T(clamping: Int16.min) ... T(clamping: Int16.max) {
    let expected = (value > 0) && (value.nonzeroBitCount == 1)
    let actual = value.isPower(of: 2)
    expectEqual(expected, actual, "(\(value)).isPower(of: 2)")
  }
```

(The copyright year at the top of this file could be updated.)

Reply:
Yes, this is great in coverage. However, checking all 16-bit possibilities seems to be a big thing at runtime, considering this is executed for a number of different `T`s. How about narrowing it from -16 to 16?

Reply:
Testing all `Int16` values seemed fast enough when I tried it. But maybe use a range which tests all possible `Int8` and `UInt8` values at least?

Can more powers of 10 also be tested in a similar way? And some more tests for each of the slow paths?

(Not only to test your current implementation, but also to test any future changes.)

Reply:
I updated the test to include more comprehensive cases. It is able to test ranges of integers with configurable base values, which is good for testing future changes.